registration/include/pcl/registration/correspondence_estimation_lookup_table.h

```diff
@@ -0,0 +1,517 @@
+
+        /** \brief Gets the number of queries performed on the lookup table. */
+        inline size_t
+        getNumberOfQueriesOnSearchTree () { return (number_of_queries_on_search_tree_); }
+
+        /** \brief Resets the number of queries performed on the lookup table. */
+        inline void
+        resetNumberOfQueriesOnSearchTree () { number_of_queries_on_search_tree_ = 0; }
+
+        /**
+          * \brief Computes the lookup table minimum and maximum bounds with the additional margin that was set previously.
+          * @param[in] pointcloud Point cloud with the data.
+          * @return True if the computed bounds are valid.
+          */
+        virtual bool
+        computeLookupTableBounds (const pcl::PointCloud<PointT>& pointcloud);
+
+        /**
+          * \brief Initialize the lookup table using the provided tree.
+          * @param[in] tree Search tree with the point data.
+          * @return True if the lookup table was initialized successfully.
+          */
+        virtual bool
+        initLookupTable (typename pcl::search::Search<PointT>::ConstPtr tree);
+
+        /**
+         * Gets the correspondence cell index associated with the query_point.
+         * @param[in] query_point Coordinates of the query point.
+         * @param[out] correspondence_index The computed index of the lookup table cell.
+         * @param[out] correspondence_index_components The computed index [x,y,z] components.
+         * @return True if the index is valid. False if the query point was outside of the lookup table bounds.
+         */
+        bool
+        computeCorrespondenceCellIndex (const PointT& query_point, size_t& correspondence_index, Eigen::Vector3i& correspondence_index_components);
+
+        /**
+         * Computes the correspondence cell centroid from the lookup table cell index components (retrieved with @see computeCorrespondenceCellIndex).
+         * @return The centroid of the correspondence cell.
+         */
+        Eigen::Vector3f
+        computeCorrespondenceCellCentroid (const Eigen::Vector3i& correspondence_index_components);
+
+        /**
+         * Computes the squared distance between two points.
+         * @param a First point.
+         * @param b Second point.
+         * @return The Euclidean squared distance between the two points.
+         */
+        float
+        computeSquaredDistance (const Eigen::Vector3f& a, const Eigen::Vector3f& b)
+        {
+          return ((a - b).squaredNorm ());
+        }
+
+        /**
+         * Computes the squared distance between two points.
+         * @param a First point.
+         * @param b Second point.
+         * @return The Euclidean squared distance between the two points.
+         */
+        float
+        computeSquaredDistance (const PointT& a, const PointT& b)
+        {
+          float dx = a.x - b.x;
+          float dy = a.y - b.y;
+          float dz = a.z - b.z;
+          return (dx * dx + dy * dy + dz * dz);
+        }
```

Comment:
Delete the method and just use `(a.getVector3fMap () - b.getVector3fMap ()).squaredNorm ()`. This is not worth to have in the public interface of the class.